src/jit/flowgraph.cpp

```diff
@@ -14465,6 +14465,10 @@ bool Compiler::fgOptimizeBranchToNext(BasicBlock* block, BasicBlock* bNext, Basi
             LIR::Range& blockRange = LIR::AsRange(block);
             GenTree*    jmp        = blockRange.LastNode();
             assert(jmp->OperIsConditionalJump());
+            if (jmp->OperGet() == GT_JTRUE)
+            {
+                jmp->gtOp.gtOp1->gtFlags &= ~GTF_SET_FLAGS;
```

Comment:
How do we end up with `GTF_SET_FLAGS` on a `JTRUE` in the first place?

Reply:
The `GTF_SET_FLAGS` isn't on the `JTRUE`, it's on the compare (its op1). It needs to be reset so that we can delete the dead code once we've removed the branch.

Reply:
Sorry, somehow I missed `gtOp.gtOp1`. I think I'd prefer `gtGetOp1()` be used instead, for some reason it stands out better.